app/assets/javascripts/app/form-validation.js

```diff
@@ -59,6 +59,7 @@ const validate = {
     f.setAttribute('aria-describedby', `alert_${f.id}`);
 
     if (f.validity.valueMissing) f.setCustomValidity(this.msgs.missing);
+    else if (f.validity.patternMismatch) f.setCustomValidity(this.msgs.mismatch);
     else if (f.validity.typeMismatch) f.setCustomValidity(this.msgs.mismatch);
```

Comment:
maybe combine these two? `else if (f.validity.patternMismatch || f.validity.typeMismatch) f.setCustomValidity(this.msgs.mismatch);`